### PR TITLE
add include parent flag validation

### LINKED ISSUE #5670: Add validation for parent flag
1. If input dataset is "RAW" tier, there is no parent files for input files so parent flag cannot be set true. 
2. If there is no input dataset (MC), parent flag cannot be set true.

### PR DESCRIPTION
**Pull request by ticoann**
fixes #5670

### CODE REVIEW

**Comment by ticoann** [comment]
@amaltaro, Alan, please review

**Comment by amaltaro** [comment]
Giving a second thought, maybe the best validation would be to make a DBS call for the parent of the input dataset. If it's not available, then raise an exception.
We may also change split the error message in 2 cases:
 - no InputDataset and IncludeParents=True -> use case not allowed
 - InputDataset with no parents and IncludeParents=True -> Parent inexistent for this input.

**Comment by ticoann** [comment]
Thanks Alan for the review. I updated the code to have 2 error cases.
I intentionally not use DBS call for validation. (If possible I would like to avoid it). That being said, is there a case which parent file not exist except "RAW" dataset?

**Comment by amaltaro** [comment]
Using DBS call would make it more generic and free of data tiers hard coded.
However, 'RAW' data tier should cover these cases. Of course the requestors can always put some silly data there (like GEN-SIM), but the probability for that to happen is almost 0 :-) So we should be fine.
Thanks!

**Comment by ticoann** [comment]
Ah, I didn't know there is GEN-SIM case as well. Anyway, I change the code call dbs as you suggested initially. I thought there was only RAW case. If there is more than one there could be more created later. I was just trying to avoid http(s) call if possible. But as you said using dbs is more general. Thanks,

File: test/python/WMCore_t/WMSpec_t/StdSpecs_t/StdBase_t.py [hunk 1]
```diff
@@ -42,12 +43,39 @@ def testStdBaseValidation(self):
...
+        
+        arguments["IncludeParents"] = True
+        arguments["InputDataset"] = "/Cosmics/Commissioning2015-6Mar2015-v1/RECO"
+        stdBaseInstance.factoryWorkloadConstruction("TestWorkload", arguments)
+        
+        arguments["IncludeParents"] = False
+        arguments["InputDataset"] = "/Cosmics/Commissioning2015-6Mar2015-v1/RECO"
+        stdBaseInstance.factoryWorkloadConstruction("TestWorkload", arguments)
+        return
+        return
```
Comment (amaltaro) [inline test/python/WMCore_t/WMSpec_t/StdSpecs_t/StdBase_t.py]: double return here :-)

Comment (ericvaandering) [inline test/python/WMCore_t/WMSpec_t/StdSpecs_t/StdBase_t.py]: Best to make sure....

**Comment by amaltaro** [comment]
GEN-SIM was just an example that some crazy person could set for the input :-)
It looks good to me now! Maybe you could just fix a typo in "_varifyDBSCall", it should be "_verifyDBSCall" :-)

File: test/python/WMCore_t/WMSpec_t/StdSpecs_t/StdBase_t.py (hunk 1 above)
Comment (ticoann) [inline test/python/WMCore_t/WMSpec_t/StdSpecs_t/StdBase_t.py]: Yes, sometimes, it doesn’t listen when I just tell that one time. :-)

On Mar 12, 2015, at 9:55 AM, Eric Vaandering <notifications@github.com<mailto:notifications@github.com>> wrote:


In test/python/WMCore_t/WMSpec_t/StdSpecs_t/StdBase_t.py<https://github.com/dmwm/WMCore/pull/5676#discussion_r26308291>:

> +        arguments["IncludeParents"] = True
> +        arguments["InputDataset"] = "/Cosmics/Commissioning2015-v1/RAW"
> +        self.assertRaises(WMSpecFactoryException, stdBaseInstance.factoryWorkloadConstruction, "TestWorkload", arguments)
> +
> +        arguments["IncludeParents"] = True
> +        self.assertRaises(WMSpecFactoryException, stdBaseInstance.factoryWorkloadConstruction, "TestWorkload", arguments)
> +
> +        arguments["IncludeParents"] = True
> +        arguments["InputDataset"] = "/Cosmics/Commissioning2015-6Mar2015-v1/RECO"
> +        stdBaseInstance.factoryWorkloadConstruction("TestWorkload", arguments)
> +
> +        arguments["IncludeParents"] = False
> +        arguments["InputDataset"] = "/Cosmics/Commissioning2015-6Mar2015-v1/RECO"
> +        stdBaseInstance.factoryWorkloadConstruction("TestWorkload", arguments)
> +        return
> +        return


Best to make sure....

**Comment by ticoann** [comment]
Thanks Alan, I fixed the typo, and removed one return statement. I will merge that. 
The only case I am concerning is whether there is case, (I added input dataset validation as well)
request is made before the input dataset is recorded in dbs (or before parent dataset is recorded.)

**Comment by amaltaro** [comment]
Not a problem, relax ;-)
PPD should not put requests into the system before their input datasets get announced.
The WF team can always make mistakes and v++, so we should be safe.

**Pull request merged**